### PR TITLE
add controller-admin branch for admin

### PR DESCRIPTION
**Pull request by HammamAbuShehadeh**
#12 add admin controller

### CODE REVIEW

**Comment by HammamAbuShehadeh** [comment]
Done Update scheme models files

File: server/server/controllers/adminController.js
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js:1]: looks good

File: server/server/controllers/adminController.js [hunk 1]
```diff
@@ -1 +1,114 @@
...
+      description,
+      dishes: [...dishes],
+    });
+
+    await newRestaurant.save();
+
+    res
+      .status(201)
+      .json({ message: "Restaurant added successfully", newRestaurant });
+      console.error("Error adding restaurant:", error);
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js]: Transfer This line of code under catch error (line 36)

File: server/server/controllers/adminController.js [hunk 2]
```diff
@@ -1 +1,114 @@
...
+    });
+    // Save the new Dish to the database
+    await newDish.save();
+    // Add the new Dish to the restaurant's dishes array
+    restaurant.dishes.push(newDish);
+    // Save the updated restaurant with the new dish reference
+    await restaurant.save();
+    res
+      .status(201)
+      .json({ newDish });
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js]: add message "Dish added to the restaurant successfully"

File: server/server/controllers/adminController.js [hunk 3]
```diff
@@ -1 +1,114 @@
...
+// get dishes for one Restaurant
+const getAllDishes = async (req, res) => {
+  try {
+    const { restaurantId } = req.params;
+
+    // Find the restaurant by its ID
+    const restaurant = await RestaurantModel.findById(restaurantId);
+
+    if (!restaurant) {
+      return res.status(404).json({ message:"there is no retaraurant with that id " });
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js]: Edit this line of code to respond error.message

File: server/server/controllers/adminController.js
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js:1]: looks good

File: server/server/controllers/adminController.js [hunk 4]
```diff
@@ -101,12 +101,35 @@ const getAllDishes = async (req, res) => {
   }
 };
 
+// delete Restaurant or Restaurants
+const deleteRestaurant = async (req, res) => {
+  try {
+    // Example: { ids: ['id1', 'id2', 'id3'] }
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js:107]: Delete this unnecessary comment

File: server/server/controllers/adminController.js [hunk 5]
```diff
@@ -120,8 +120,62 @@ const deleteRestaurant = async (req, res) => {
...
+};
 
+// for search dishes minPrice and maxPrice and category and name
+const searchDishes = async (req, res) => {
+  try {
+    const { minPrice, maxPrice, category, name } = req.query;
+
+    // Define the search criteria
+    const searchCriteria = {};
+    if (minPrice != undefined && !isNaN(minPrice)) {
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js]: I think it must be two equal here like this !==

File: server/server/controllers/adminController.js [hunk 6]
```diff
@@ -120,8 +120,62 @@ const deleteRestaurant = async (req, res) => {
...
+      searchCriteria.price = { $gte: minPrice };
+    }
+    if (maxPrice != undefined && !isNaN(maxPrice)) {
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js]: here also !==

File: server/server/controllers/adminController.js [hunk 7]
```diff
@@ -212,6 +212,43 @@ const editRestaurant = async (req, res) => {
   }
 };
 
+// edit dish
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js:215]: change comment To Edit Dish Information

File: server/server/controllers/adminController.js [hunk 8]
```diff
@@ -249,7 +249,22 @@ const editDish = async (req, res) => {
   }
 };
 
-
+// Get past completed orders for report 
+//for test
```
Comment (FarahAlsoqi) [inline server/server/controllers/adminController.js]: remove this comment

**Review by FarahAlsoqi — COMMENTED** [review]
thx great work

File: server/server/controllers/adminController.js
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js:1]: thanks hammam

File: server/server/controllers/adminController.js (hunk 1 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js]: ok

File: server/server/controllers/adminController.js (hunk 2 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js]: yes sir

File: server/server/controllers/adminController.js (hunk 3 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js]: on my way to do it.

File: server/server/controllers/adminController.js
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js:1]: thanks very much.

File: server/server/controllers/adminController.js (hunk 4 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js:107]: yep yep

File: server/server/controllers/adminController.js (hunk 5 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js]: yes my bad.

File: server/server/controllers/adminController.js (hunk 6 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js]: kk

File: server/server/controllers/adminController.js (hunk 7 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js:215]: تكرمي

File: server/server/controllers/adminController.js (hunk 8 above)
Comment (AHMADJUAM) [reply server/server/controllers/adminController.js]: kk

**Pull request merged**